learning_observer/learning_observer/learning_tools/schoology.py

```diff
@@ -0,0 +1,152 @@
+import json
+import requests
+import configparser
+from requests_oauthlib import OAuth1
+
+class Schoology:
+    def __init__(self, config_path='./config.ini'):
+        self.config = configparser.ConfigParser()
+        self.config.read(config_path)
+        API_KEY = self.config['SCHOOLOGY_CONFIG']['API_KEY']
+        SECRET = self.config['SCHOOLOGY_CONFIG']['SECRET']
+        self.auth = OAuth1(API_KEY, SECRET, "", "")
+        self.base_url = f'https://api.schoology.com/v1/'
+
+    # REST Methods
+    def _get(self,uri):
+        return requests.get(uri, auth=self.auth)
+
+    def _post(self,uri,data):
+        return requests.post(uri, json=data, auth=self.auth)
+
+    def _put(self,uri,data):
+        return requests.post(uri, json=data, auth=self.auth)
+
+    def _delete(self,uri):
+        return requests.delete(uri, auth=self.auth)
+    
+    def api_call(self, method, endpoint, params=None, data=None, absolute_url=False, retry=True):        
+        if absolute_url:
+            url = endpoint
+        else:
+            url = self.base_url + endpoint
+            # Append params key/values if added
+            if params:
+                url += '?' + '&'.join(f"{k}={v}" for k, v in params.items())
+
+        response_func = getattr(requests, method.lower())
+        response = response_func(url, auth=self.auth, params=params, data=json.dumps(data) if data else None)
+        
+        if response.status_code != 200:
+            response.raise_for_status()
+        
+        return response.json()
+
+
+    # Schoology API calls
+
+    #can be used to retrieve user information if email of the student is known
+    def get_user_id(self,email,users=None):
```

Comment:
Clean up documentation. Use docstrings under the header instead of single line comments above it.